### ISSUE TITLE
The app hangs if the user tries to add the Book on rows where it is missing

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord/tc-create-app#639]
DoD: Users are able to translate deleted rows or translate new content. 

=============
v1.1.0-rc.3
The file is at: https://qa.door43.org/translate_test/en_tn/_edit/tc01-tc-create-1/en_tn_55-1TI.tsv
The book, chapter, verse are missing because the target has fewer rows than the source.
1. Load the file
2. Scroll to near the end where the parallel scripture pane is blank and the book, chapter, verse are also blank
3. Add the book abbreviation and click out of the field
4. Note the gray screen and frozen app

https://images.zenhubusercontent.com/191973535/8b529aea-49d4-4fdf-8f7a-e5e498d5df36/639.mp4

**Comment by mandolyte** [comment]
@cckozie @birchamp 

When I did the above, I see that the chapter and verse are for Titus 2:5, and it follows Titus 3:15 and is the last tN row in the file.

This leads me to guess that it may be a deleted row in the target. It would make sense (and I seem to recall something like this) that deleted rows in target are placed at the end of the source. So is this just a normal deleted target row?

If it is, how is one supposed to re-insert a deleted row?

![image](https://user-images.githubusercontent.com/1289437/111157404-39a55c80-856d-11eb-858e-5c6d84b995b9.png)

**Comment by mandolyte** [comment]
I did try a deleted (having never done one before) and saw a warning that there is no "undo" ... so it is permanent I guess.

**Comment by mandolyte** [comment]
Sorry for all the comments... just keep thinking of things to try. I opened 3JN and deleted the very first row. It disappeared from the screen, but I found it at the end. And if I tried to edit anything on the target side, the same gray screen of death appears.

So perhaps the solution is to not show any deleted rows so this kind of attempt cannot be made?

@ancientTexts-net @cckozie

**Comment by cckozie** [comment]
Katalon tested in tCC Test App Crashes

**Comment by birchamp** [comment]
@mandolyte please note that this is slated for 1.2

**Comment by ancientTexts-net** [comment]
Sample from testing:

[https://app.zenhub.com/files/191973535/5dc19ae7-0c98-4607-9a4f-6cce7dcea384/download](https://app.zenhub.com/files/191973535/5dc19ae7-0c98-4607-9a4f-6cce7dcea384/download)

**Comment by cckozie** [comment]
v1.2.0-rc.1 build 26-371f3c3
The app no longer hangs when adding target data to empty target rows. However, the apps behavior is less than ideal and will likely cause confusion to users. Here are the details of my test:
I began with 4 "missing" rows in the target. I'll call them rows A, B, C, and D and they are displayed in that order at the end of the file even though they really belong close to the top.
I added the book abbreviation to row B and clicked out of the field.
The app moved row B up and out of view, and is now showing row A where B used to be.
The app filled in the other fields in row B copying the data from the source.
The rows are now in B, A, C, D order in the app, still at the end of the file.
The row remains at the end of the original non-missing rows even after saving and closing and restarting the app.

The following changes would make the app much more user friendly:
1. Do not scroll the row being edited out of view after the user makes a change in it
2. Re-sort the row into the proper sequence when the Save button is clicked
